Automattic/wp-calypso
# update: a4a improve mobile layout

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/155

## Proposed Changes
Update the layout so it will look responsive on mobile devices. There are a number of changes which will be discussed in a post: pfunGA-OV-p2
Some notes:
- Sites page is a bit broken and contains temporary code, so I did only minimal updates.
- There are a number of code duplications through this PR, but we could address these separately during refactoring after we are happy with the look.
- I used `useBreakpoint( '<660px' )` instead `useMobileBreakpoint` as 660 is widely used through codebase, and it allows to keep things consistent.

| Overview page | Marketplace | Licenses |
| ---- | ---- | ---- |
| <img width="407" alt="Screenshot 2024-03-28 at 4 00 44 PM" src="https://github.com/Automattic/wp-calypso/assets/60262784/77d0c8ee-f8e6-4ac2-b049-9d7ea03e5077"> | <img width="408" alt="Screenshot 2024-03-28 at 4 01 47 PM" src="https://github.com/Automattic/wp-calypso/assets/60262784/37fc3113-ba55-4e35-b61a-e72af2341eff"> | <img width="407" alt="Screenshot 2024-03-28 at 4 02 26 PM" src="https://github.com/Automattic/wp-calypso/assets/60262784/2fe24387-e587-4cad-bd84-e0b0753b553b"> |


## Testing Instructions



- Spin up the branch locally
- Check different pages on mobile device

## Pre-merge Checklist
